js/GroupedBarChart.js

```diff
@@ -122,87 +130,126 @@ var GroupedBarChart = function(param)
          * @param data - parsed data from the input json
          */
-        initChart : function(data) {
-          var that = this;
-
-          that.setScales(data);
-
-          svg.append("g")
-            .attr("class", "y axis")
-            .call(that.yAxis);
-
-          svg.append("g")
-            .attr("class", "x axis")
-            .attr("transform", "translate(0," + that.h + ")")
-            .call(that.xAxis);
+        initChart : function(data, precision) {
+            var that = this;
+            that.setScales(data);
+            that.precision = precision;
+
+            svg.append("g")
+                .attr("class", "y axis")
+                .call(that.yAxis)
+            ;
+
+            svg.append("g")
+                .attr("class", "x axis")
+                .attr("transform", "translate(0," + that.h + ")")
+                .call(that.xAxis)
+            ;
         },
         /**
          * Updates the chart based on the data passed in.
          * @param data - parsed data from input json
          */
         updateChart : function(data)
         {
-          var that = this;
-
-          /* Transition Functions */
-          /* Enter the data into the dom */
-          var barsAnimationTime = 1000;
-          svg.append("g").selectAll("g")
-            .data(that.mainCategories)
-            .enter().append("g")
-              .attr("transform", function(d) { return "translate(" + that.xScale(d) + ",0)"; })
-            .selectAll("rect")
-              .data(function(d){ return _.where(data, {xval: d});})
-              .enter().append("rect")
+            var that = this;
+
+            var tooltip_mouseover = function(d) {
+                var tooltipText = '';
+                if (d.name)
+                    tooltipText = "<strong>" + d.name + "</strong>";
+                if (d.yval)
+                    tooltipText +=  "<br>Score: <strong>" + d.yval.toFixed(that.precision)  + "</strong>";
+
+                if (d.zval)
+                    tooltipText += "<br>Volume: <strong>" + d.zval.toFixed(that.precision)  + "</strong>";
+
+
+                tooltip.transition()
+                    .duration(200)
+                    .style("opacity", 1)
+                    //.style("border-color", barColorCode);
+                tooltip.html(tooltipText)
+                    .style("border-color", d3.select(this).attr("fill"))
+                    .style("background-color", "#FFFFFF")
+                    ;
+            };
+
+            var tooltip_mousemove = function(d) {
+                var mouseContainer = d3.select(elem + " svg");
+                var mouseCoords    = d3.mouse(mouseContainer.node());
+
+                d3.select(elem + " div.tooltip")
+                    .style("left", (mouseCoords[0]) + "px")
+                    .style("top",  (mouseCoords[1]) - 50 + "px");
+                if(d.zval) {
+                    d3.select(elem + " div.tooltip")
```

Comment:
you can store d3.select(elem + " div.tooltip") into a variable at the top first for efficiency reasons. but we can do that later